holoviews/core/spaces.py

```diff
@@ -590,6 +614,15 @@ def __init__(self, callback, initial_items=None, **params):
 
         super(DynamicMap, self).__init__(initial_items, callback=callback, **params)
         invalid = [s for s in self.streams if not isinstance(s, Stream)]
+
+        if isinstance(self.callback, Generator):
+            if self.kdims:
+                raise Exception('Generators can only be used without key dimensions')
+            if self.streams == []:
+                self.warning('Empty streams required to trigger generator')
```

Comment:
We still need an *empty* stream without *any* stream parameters to trigger the generator to trigger ``next`` via ``dmap.event``. Without it, nothing can change after first display.